### PR TITLE
EE-1045: Fix slashing for validators

### PR DESCRIPTION
**Pull request by mpapierski**
Ref: https://casperlabs.atlassian.net/browse/EE-1045

Removes `quash_bid` and makes it an impl detail, and then `slash` uses it to slash validator bids.

### CODE REVIEW

**Comment by EdHastingsCasperLabs** [comment]
Can you update the `should_step` test to include slashing check? There are some TODO's in the test indicating the changes that were waiting on this bug fix.

**Comment by mpapierski** [comment]
> Can you update the `should_step` test to include slashing check? There are some TODO's in the test indicating the changes that were waiting on this bug fix.

Done

File: grpc/tests/src/test/step.rs [hunk 1]
```diff
@@ -85,41 +74,33 @@ fn should_step() {
...
 
-    // TODO: this is currently not possible due to a bug in the slashing logic
-    //       the genesis validators are not slashable due to an oversight in initialization.
-    // let _bid_purses_before_slashing: BidPurses = get_value(&mut builder, bid_purse_key);
-    // assert!(
-    //     bid_purses_before_slashing.contains_key(&ACCOUNT_1_PK),
-    //     "should contain slashed validator)"
-    // );
+    let bids_before_slashing: Bids = builder.get_value(auction_hash, BIDS_KEY);
+    assert!(bids_before_slashing.contains_key(&ACCOUNT_1_PK));
```
Comment (EdHastingsCasperLabs) [inline grpc/tests/src/test/step.rs:92]: Asserts should have a comment indicating what the failure means, particularly when there are more than one asserts in the same test.

Comment (mpapierski) [reply grpc/tests/src/test/step.rs]: Done

**Review by henrytill — APPROVED** [review]
LGTM

**Review by EdHastingsCasperLabs — APPROVED** [review]

**Pull request merged**